### PR TITLE
Pycon19 - Added Sponsor Type Options

### PR DESCRIPTION
**Pull request by soachishti**
@murad-hubib Please review, merge and deploy.

I have added sponsor type field as we are categorization sponsor into Silver, Gold, Community... for PyCon 19.

### CODE REVIEW

File: theme/admin.py [hunk 1]
```diff
@@ -33,11 +33,19 @@ class GalleryAdmin(DisplayableAdmin):
     list_display = ('title', 'status')
     inlines = [GalleryImageInline]
 
+class SponsorTypeAdmin(DisplayableAdmin):
+    def get_model_perms(self, request):
```
Comment (soachishti) [inline theme/admin.py:37]: Hiding sponsor type from admin menu, it looks more suitable when inline with "Sponsor type" in sponsor admin page.

File: theme/models.py [hunk 2]
```diff
@@ -84,7 +87,7 @@ def get_absolute_url(self):
                      format="Image", max_length=255)
     link = models.URLField()
     featured = models.BooleanField(default=False)
-
+    sponsor_type = models.ForeignKey(SponsorType, null=True, blank=True, on_delete=models.SET_NULL)
```
Comment (soachishti) [inline theme/models.py:90]: Sponsor with null type will be shown at first place.

File: theme/models.py [hunk 3]
```diff
@@ -75,6 +75,9 @@ class GalleryImage(models.Model):
     def __unicode__(self):
         return self.gallery.title
 
+class SponsorType(Displayable):
+    title = models.CharField(null=True, blank=True, max_length=100)
+    order = models.IntegerField(default=0)
```
Comment (soachishti) [inline theme/models.py:80]: To order sponsor category first: Gold and second: Silver.

File: theme/templatetags/sponsor_tags.py [hunk 4]
```diff
@@ -4,12 +4,22 @@
 
 register = template.Library()
 
+def preprocess_sponsors(sponsors):
```
Comment (soachishti) [inline theme/templatetags/sponsor_tags.py:7]: Converting list of sponsor to dictionary where key is sponsor type and value is sponsors of given type.

**Comment by murad-hubib** [comment]
@soachishti Sure. Will do that tomorrow as this PR has migrations

**Comment by soachishti** [comment]
Ok, great!

**Review by murad-hubib — APPROVED** [review]

**Comment by murad-hubib** [comment]
@soachishti Done & deployed. Please verify your changes. Good work

**Comment by soachishti** [comment]
Thank you 👍

**Pull request merged**